### PR TITLE
Groundwork to enable better docker automated builds

### PR DESCRIPTION
**Pull request by milutz**
**Proposed changes**:
- Rewrite and restructuring of the docker support for RASA, to enable automatic-builds of docker containers in multiple different "flavors"
- In addition to the Dockerfile changes, a number of new subdirectories were added to reduce the cutter these changes created in the root:
  - `docker` - All the new docker files (and docker cloud button support)
  - `alt_requirements` - All the requirements files beyond the initial root `requirements.txt`(which now  includes (`-r xxx`) elements from the files in alt to avoid repeats)
  - `sample_configs` - All the sample_config files
  - `heroku` - heroku button support files (assumed obsolete)
- The `README.md` file has been restructured to have both a docker and python quick-start at the top (needed because the same readme is used for the docker hub/cloud webpages, and its confusing if there isn't some description of the docker use near the top of the file (example: https://hub.docker.com/r/samtecspg/rasa_nlu/ )
- Once this PR is merged we will make the changes to the RasaHQ docker hub/cloud to enable the automatic builds:
  - Four build complexities will be built (bare, mitie, spacy, and full) on each push to master and also for each release tag
- This is the combined work of @wrathagom @dcalvom and myself

Our testing instance contains examples the automated docker builds that this change will enable: https://hub.docker.com/r/samtecspg/rasa_nlu/tags/ (currently only shows a `latest` examples)

**Status**:
- [X] ready for code review
Code is considered complete unless changes are requested or fixes/improvements are needs to/from testing

- [X] there are tests for the functionality
Manual testing has been created via the new `docker/Dockerfile_test` Dockerfile and (when the automated builds are enabled) the automated build process will alert if there is a build error.

- [X] documentation updated
Documentation update complete

- [X] changelog updated
Changelog entries added

### CODE REVIEW

**Comment by wrathagom** [comment]
A few more notes:

- I just updated the README a bit
- In tandem with this I'll be submitting a new readme to get the `/rasa/rasa_nlu` library marked as **official** on docker hub. This will allow me to have a custom README on docker hub specific to using the docker images.

**Comment by wrathagom** [comment]
@tmbo I think that was a travis timeout?

**Comment by wrathagom** [comment]
@tmbo - @milutz and I are on standby waiting for this pull so that we can get the automated builds and `official` docker builds set up.

File: README.md [hunk 1]
```diff
@@ -9,63 +9,65 @@
 Rasa NLU (Natural Language Understanding) is a tool for intent classification and entity extraction. You can think of Rasa NLU as a set of high level APIs for building your own language parser using existing NLP and ML libraries. Find out more on the [homepage of the project](https://rasa.ai/), where you can also sign up for the mailing list.
 
 **Extended documentation:**
-- [latest](https://rasa-nlu.readthedocs.io/en/latest/)&nbsp; (if you install from **github**) or 
-- [stable](https://rasa-nlu.readthedocs.io/en/stable/) (if you install from **pypi**)
+- [stable](https://rasa-nlu.readthedocs.io/en/stable/) (if you install from **X.X.X** [docker](https://hub.docker.com/r/rasa/rasa_nlu/tags/) image or **pypi**)
+- [latest](https://rasa-nlu.readthedocs.io/en/latest/)&nbsp; (if you install from **latest** [docker](https://hub.docker.com/r/rasa/rasa_nlu/tags/() image or **github**)
```
Comment (tmbo) [inline README.md]: there seems to be an additional unecessary `(` (after the docker link)

**Comment by tmbo** [comment]
Alright, let's get this merged today if possible on your end.

There is one thing that needs to be done though before we can merge: I just noticed that the configuration files that have been moved are referenced in the documentation a couple of times. It owuld be great if you could just search for the occurences of the six files within the `docs/` folder and update the references.

And please merge in master once more, there is a conflict with one of the configuraton files.

**Comment by wrathagom** [comment]
@tmbo I think I am good now.

**Comment by tmbo** [comment]
Thank you and your team so much, this is a great addition to the project 🎆 🎉

**Comment by wrathagom** [comment]
Your welcome, I'll get to work on the automated builds.

**Pull request merged**